enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx

```diff
@@ -76,20 +82,33 @@ export const MetabaseProviderInternal = ({
   }, [store, config.metabaseInstanceUrl]);
 
   return (
-    <Provider store={store}>
-      <EmotionCacheProvider>
-        <SdkThemeProvider theme={theme}>
-          <AppInitializeController className={className} config={config}>
-            {children}
-          </AppInitializeController>
-        </SdkThemeProvider>
-      </EmotionCacheProvider>
-    </Provider>
+    <EmotionCacheProvider>
+      <SdkThemeProvider theme={theme}>
+        <SdkFontsGlobalStyles baseUrl={config.metabaseInstanceUrl} />
+        <div className={className} id={EMBEDDING_SDK_ROOT_ELEMENT_ID}>
+          <PortalContainer />
+          {children}
+        </div>
+      </SdkThemeProvider>
+    </EmotionCacheProvider>
   );
 };
 
 export const MetabaseProvider = memo(function MetabaseProvider(
   props: MetabaseProviderProps,
 ) {
-  return <MetabaseProviderInternal store={store} {...props} />;
+  return (
+    <Provider store={store}>
+      <MetabaseProviderInternal store={store} {...props} />
+    </Provider>
+  );
 });
+
+/**
+ * This is the portal container used by popovers modals etc, it is wrapped with withPublicComponentWrapper
+ * so that it has our styles applied.
+ * Mantine components needs to have the defaultProps set to use `EMBEDDING_SDK_PORTAL_CONTAINER_ELEMENT_ID` as target for the portal
+ */
+const PortalContainer = withPublicComponentWrapper(() => (
+  <div id={EMBEDDING_SDK_PORTAL_CONTAINER_ELEMENT_ID}></div>
+));
```

Comment:
This looks a bit overkill, well not by much, but hear me out.

`withPublicComponentWrapper` wraps whatever's inside with, obviously, a `PublicComponentWrapper`, and passes through whatever props from this component to the component inside, since we don't pass any prop here, I think using `PublicComponentWrapper` here might make the intent a bit clearer.

```suggestion
const PortalContainer = () => (
  <PublicComponentWrapper>
    <div id={EMBEDDING_SDK_PORTAL_CONTAINER_ELEMENT_ID}></div>
  </PublicComponentWrapper>
);
```

Another benefit is that I don't have to go inside this HoC util to figure out what it does.